synapse/http/server.py

```diff
@@ -459,7 +453,7 @@ def _send_error_response(
         request: SynapseRequest,
     ) -> None:
         """Implements _AsyncResource._send_error_response"""
-        return_json_error(f, request)
+        return_json_error(f, request, None)
```

Comment:
I've made this an explicit choice rather than an optional. I couldn't figure out a non-invasive way of providing `hs` to this class so instead here we just don't support unstable error codes on anything directly subclassing `DirectServeJsonResource`.

A cursory glance at the code seems to indicates this isn't used by any REST API classes that interact with the auth parts of Synapse, so it's unlikely we would raise an unstable error in those parts. Most APIs directly or indirectly subclass  `JsonResource` which does work for us.